vic/drivers/cesm/cpl_mct/lnd_comp_mct.F90

```diff
@@ -0,0 +1,722 @@
+    TYPE(seq_cdata)             , INTENT(inout) :: cdata_s
+    TYPE(mct_aVect)             , INTENT(inout) :: x2s, s2x
+    CHARACTER(len=*), OPTIONAL  , INTENT(in)    :: NLFilename
+
+    !--- Local Variables
+    INTEGER                                     :: LNDID
+    INTEGER                                     :: mpicom_lnd
+    INTEGER                                     :: mytask, ierr
+    INTEGER                                     :: lsize, gsize
+    INTEGER                                     :: nx_global, ny_global
+    INTEGER                                     :: dtime  ! timestep (seconds)
+    INTEGER, POINTER                            :: idata(:)
+    INTEGER, DIMENSION(:), ALLOCATABLE          :: gindex  ! Number the local grid points
+    REAL(r8), POINTER                           :: lon_data(:)
+    REAL(r8), POINTER                           :: lat_data(:)
+    REAL(r8), POINTER                           :: area_data(:)
+    REAL(r8), POINTER                           :: mask_data(:)
+    REAL(r8), POINTER                           :: frac_data(:)
+    TYPE(mct_gsMap),  POINTER                   :: GSMap_lnd
+    TYPE(mct_gGrid),  POINTER                   :: dom_lnd
+    TYPE(seq_infodata_type), POINTER            :: infodata
+    CHARACTER(len=VICMAXSTRING)                 :: caseid
+    CHARACTER(len=VICMAXSTRING)                 :: starttype
+    CHARACTER(len=VICMAXSTRING)                 :: calendar
+    LOGICAL                                     :: exists  ! true if file exists
+    CHARACTER(len=*), PARAMETER                 :: subname = 'lnd_init_mct'
+    CHARACTER(len=*), PARAMETER                 :: FORMAT = "('("//TRIM(subname)//") :',A)"
+
+    !--- Local Variables for C/Fortran Interface
+    INTEGER(C_INT)                              :: errno
+    CHARACTER(len=VICMAXSTRING, kind=C_CHAR)    :: vic_global_param_file
+
+    !--- initialize the field index values, subroutine is below in this file
+    CALL cpl_indices_set()
+
+    !--- point/get data from cdata datatype
+    CALL seq_cdata_setptrs(cdata, ID=LNDID, mpicom=mpicom_lnd, &
+         gsMap=GSMap_lnd, dom=dom_lnd, infodata=infodata)
+    CALL MPI_COMM_RANK(mpicom_lnd, mytask, ierr)
+
+    !--- copy/hand the mpicom from the driver to the vic model
+    errno = vic_cesm_init_mpi(mpicom_lnd)
+    IF (errno /= 0) THEN
+       CALL shr_sys_abort(subname//':: vic_cesm_init_mpi returned a errno /= 0')
+    ENDIF
+
+    !--- get unit number for vic log file
+    !--- setup vic log file
+    !--- set shr unit number to vic log file
+    CALL shr_file_getLogUnit(shrlogunit)
+    IF (mpicom_lnd == 0) THEN
+       ! Tony look here INQUIRE(file='lnd_modelio.nml'//TRIM(inst_suffix), exist=exists)
+       exists = .FALSE.
+       IF (exists) THEN
+          iulog = shr_file_getUnit()
+          !Tony look here CALL shr_file_setIO('lnd_modelio.nml'//TRIM(inst_suffix), iulog)
+          CALL shr_file_setIO('lnd_modelio.nml', iulog)
+       ENDIF
```

Comment:
@apcraig - We are creating per task log files inside VIC.  Do we need this section?  Should we have a separate log file for the driver?  I think it is pretty difficult to share file handles between C and Fortran.